MonoGame.Framework/Vector3.cs

```diff
@@ -522,7 +530,7 @@ public bool Equals(Vector3 other)
         /// <returns>Hash code of this <see cref="Vector3"/>.</returns>
         public override int GetHashCode()
         {
-            return (int)(this.X + this.Y + this.Z);
+            return X.GetHashCode() + Y.GetHashCode() + Z.GetHashCode();
```

Comment:
@Shqrdx - Just a heads up on what is going on.  We're reviewing all the code you've submitted for potential violations of copyright.  After we finish this we'll let you know what we have found.

Once we get thru all that we can see how you can contribute moving forward.  I think we can work something out.